### PR TITLE
CSS-10232 Create Trino users from juju secret

### PR DESCRIPTION
**Pull request by AmberCharitos**
This PR provides a mechanism for trino to load multiple users into the user `password.db`, via juju user secrets.

Merge blocked by PR, which implements the necessary library in the rock: https://github.com/canonical/charmed-trino-rock/pull/8, this should resolve the integration test errors once the rock is updated.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -319,17 +329,53 @@ def set_java_truststore_password(self, event):
...
+        """
+        try:
+            secret = self.model.get_secret(id=secret_id)
+            content = secret.get_content(refresh=True)
+        except SecretNotFoundError:
+            logger.error(f"secret {secret_id!r} not found.")
+            raise
+        return content
+
+    def _enable_password_auth(self, event):
```
Comment (gtato) [inline src/charm.py:352]: np: i found the naming of this a bit confusing.. I was wondering why are we enabling the auth every time a config changes... wasn't it enabled. I don't have a good name in mind but maybe smth like `_update_password_db`?

**Review by gtato — APPROVED** [review]
lgtm

File: README.md [hunk 2]
```diff
@@ -97,6 +97,30 @@ In order to connect clustered database systems to Trino please connect the read-
...
+juju config trino-k8s google-client-secret=<secret>
+
+```
+
+### User/password
+Additionally user/password authentication can be enabled via a Juju secret.
+
+```
+# Create the secret and grant access to Trino.
+juju add-secret trino-user-management --file /path/to/user-secrets.yaml
```
[the remark this answers is not in the document]
Comment (AmberCharitos) [reply README.md:116]: added :)

File: src/charm.py (hunk 1 above)
Comment (AmberCharitos) [reply src/charm.py]: Yeah I found it hard to name.. I like your suggestion better and haven't been able to think of a more appropriate one, so using that.

**Review by mertalpt — APPROVED** [review]

**Review by kelkawi-a — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -97,6 +97,37 @@ In order to connect clustered database systems to Trino please connect the read-
 salesforce #read-write endpoint
 salesforce_ro #read-only endpoint
 ```
+## User management
+By default authentication is not enabled for Trino. Trino supports multiple forms of authentication mechanisms to be implements at the same time. Available with the charm are Google Oauth and user/password authentication. We recommend user/password for application users which do no support Oauth, and Oauth for everything else.
```
[the remark this answers is not in the document]
Comment (AmberCharitos) [reply README.md]: Thanks, updated

File: src/utils.py [hunk 4]
```diff
@@ -228,3 +212,38 @@ def add_cert_to_truststore(container, name, cert, storepass, conf_path):
...
+            "htpasswd",
+            "-b",
+            "-B",
+            "-C",
+            "10",
+            db_path,
+            user,
+            password,
+        ]
+        db_exists = container.exists(db_path)
```
[the remark this answers is not in the document]
Comment (AmberCharitos) [reply src/utils.py]: Yes it is, updated.

**Pull request merged**